numpy/core/arrayprint.py

```diff
@@ -294,8 +294,9 @@ def printoptions(*args, **kwargs):
     Examples
     --------
 
+    >>> from numpy.testing import assert_equal
     >>> with np.printoptions(precision=2):
-    ...     print(np.array([2.0])) / 3
+    ...     np.array([2.0]) / 3
     [0.67]
```

Comment:
This surprises me - shouldn't it be showing the `repr` here?

Reply:
The true result is `array([0.67])` if that's what you mean, but refguide_check has a number of fallback switches and seems to be "smart enough" to pass with just `[0.67]`.

I'm changing to the more formally correct result as you imply here, but probably won't go hunting for these cases too actively for now since some leniency on the "printed" values is one of the reasons for picking refguide over vanilla doctest.

Reply:
@eric-wieser it basically does `assert_allclose(eval(want), eval(got))`